server/client.go

```diff
@@ -3935,6 +3939,13 @@ func (c *client) processInboundClientMsg(msg []byte) (bool, bool) {
 		c.pubPermissionViolation(c.pa.subject)
 		return false, true
 	}
+	// Check if the client is trying to publish to reserved NRG subjects.
+	// Doesn't apply to NRGs themselves as they use SYSTEM-kind clients instead.
+	if c.kind == CLIENT && bytes.HasPrefix(c.pa.subject, clientNRGPrefix) && c.acc != c.srv.SystemAccount() {
```

Comment:
Doe we grab server lock for SystemAccount? If so are we violating locking order?

Reply:
Good point, it does take the lock, but `acc` is already taken under the client lock earlier too, so moved this down to after the client unlock.